### PR TITLE
fix: failed explorer upgrade in migration due to pod name change

### LINKED ISSUE #3207: Migration test failed with explorer 0.26.0.0 : no message received through API query (create-topic.js)
## Describe the bug
Migration test failed in CI when running scripts/create-topic.js. The script reports no message received through the API query and then throws an error.

## Expected behavior
Migration test should receive a message through the API query and complete successfully.

## Steps to reproduce
1. Run the migration test job in CI (or run scripts/create-topic.js in the migration test flow).
2. Observe failure after retries expire.

## Logs
```
[2026-01-26T20:39:18.995Z] ERROR: ❌ ERROR: No message received through API query (retries: 90 of 90)
[2026-01-26T20:39:18.996Z] ERROR: ❌ ERROR: Error: ❌ ERROR: Test failed due to issues detected. Error: ❌ ERROR: Test failed due to issues detected.
    at main (file:///home/runner/_work/solo/solo/scripts/create-topic.js:336:13)
node:internal/process/promises:394
    triggerUncaughtException(err, true /* fromPromise */);
    ^

Error: ❌ ERROR: Test failed due to issues detected.
    at main (file:///home/runner/_work/solo/solo/scripts/create-topic.js:336:13)
```

## Environment
- Date: 2026-01-26
- Runner: GitHub Actions (migration test job)

## Additional context
If helpful, I can provide full job logs or artifacts.

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* update k8 client to check pod status only existing pod not for pod in shut down stage
* for explorer, upgrade to chart trigger a pod with different name, must stop portfoward first then re-eanble

### Related Issues

* Closes #3207 
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: src/integration/kube/k8-client/resources/pod/k8-client-pod.ts [hunk 1]
```diff
@@ -329,7 +329,7 @@ export class K8ClientPod implements Pod {
     return v1Pod;
   }
 
-  public static fromV1Pod(v1Pod: V1Pod, pods: Pods, coreV1Api: CoreV1Api, kubeConfig: KubeConfig): Pod {
+  public static fromV1Pod(v1Pod: V1Pod, pods: Pods, coreV1Api: CoreV1Api, kubeConfig: KubeConfig): K8ClientPod {
```
Comment (jeromy-cannon) [inline src/integration/kube/k8-client/resources/pod/k8-client-pod.ts]: We cannot expose the K8Client* instances.  That defeats the purpose of the interfaces.

Comment (JeffreyDallas) [reply src/integration/kube/k8-client/resources/pod/k8-client-pod.ts]: resolved

**Review by jeromy-cannon — APPROVED** [review]
